src/main/java/no/finn/unleash/util/IpAddressMatcher.java

```diff
@@ -52,14 +54,14 @@ public IpAddressMatcher(String ipAddress) {
         }
     }
 
-    public boolean matches(String address) {
+    public boolean matches(@Nullable String address) {
         if (address == null || address.isEmpty() || requiredAddress == null) {
             return false;
         }
 
         InetAddress remoteAddress = parseAddress(address);
 
-        if (!requiredAddress.getClass().equals(remoteAddress.getClass())) {
+        if (remoteAddress == null || !requiredAddress.getClass().equals(remoteAddress.getClass())) {
```

Comment:
Annotations uncovered that `parseAddress` returns a `null` in some cases. Added the check to account for that. Please check the logic is correct.